src/main/kotlin/racing/view/input/ConsoleInputView.kt

```diff
@@ -0,0 +1,84 @@
+package racing.view.input
+
+internal class ConsoleInputView : InputView {
+    override fun displayCarNamesQuestion() {
+        println("경주할 자동차 이름을 입력하세요(이름은 쉼표(,)를 기준으로 구분).")
+    }
+
+    override fun displayAttemptCountQuestion() {
+        println("시도할 횟수는 몇 회인가요?")
+    }
+
+    override fun inputCarNames(input: String?): List<String> {
+        val validatedInput = validateCarNamesInput(input)
+        return validatedInput.split(DELIMITER_COMMA).map { it.trim() }
+    }
+
+    override fun inputNumber(input: String?): Int {
+        return validateIntInput(input).toInt()
+    }
+
+    override fun promptForCharNames(questionBlock: () -> Unit): List<String> {
+        questionBlock.invoke()
+        val inputCarNames = validateAndGetCarNames(questionBlock = questionBlock)
+        return inputCarNames
+    }
+
+    override fun promptForNumberInput(questionBlock: () -> Unit): Int {
+        questionBlock.invoke()
+        val inputNumber = validateAndGetNumber(questionBlock = questionBlock)
+        return inputNumber
+    }
+
+    override fun promptAndValidateCarNamesInput(): List<String> {
+        return promptForCharNames(::displayCarNamesQuestion)
+    }
+
+    override fun promptAndValidateAttemptCountInput(): Int {
+        return promptForNumberInput(::displayAttemptCountQuestion)
+    }
+
+    private fun validateAndGetCarNames(
+        input: String? = readlnOrNull(),
+        questionBlock: () -> Unit,
+    ): List<String> {
+        return try {
+            inputCarNames(input)
+        } catch (e: IllegalArgumentException) {
+            e.printStackTrace()
+            questionBlock.invoke()
```

Comment:
아래와 같이도 작성 가능해요!! :)
```suggestion
            questionBlock()
```

Reply:
구현체 내에서 메서드 참조가 아니라 람다식의 호출임이 명확하게 드러나는 것 같아서 invoke를 유지하고 있었습니다.
현재 진행하는 과제들의 분량은 많은 편이 아니니 invoke는 생략해도 괜찮을것 같습니다!
수정하였습니다.